src/containers/blocks/AddURLBlock.js

```diff
@@ -0,0 +1,85 @@
+import React, {Component} from 'react';
+import {connect} from "react-redux";
+import {bindActionCreators} from 'redux';
+import {addUrlToTest, hideUrlInput, toggleUrlInput, removeUrlFromTest} from "../../actions"
+
+class AddURLBlock extends Component {
+  constructor(props){
+    super(props);
+
+    this.state = {
+      urlValue: ""
+    }
+
+    this.handleInputChange = this.handleInputChange.bind(this);
+  }
+
+  componentDidMount() {
+    this.props.hideUrlInput();
+  }
+
+  handleAddUrl() {
+    this.props.toggleUrlInput();
+  }
+
+  handleInputChange(value) {
+    this.setState({
+      ...this.state,
+      urlValue: value
+    });
+  }
+
+  handleUrlAdded() {
+    if(this.state.urlValue.length > 0) {
+      this.props.addUrlToTest(this.state.urlValue);
+      this.setState({
+        ...this.state,
+        urlValue: ""
+      });
+      this.props.toggleUrlInput();
+    }
+  }
+
+  handleUrlDeletion(url) {
+    this.props.removeUrlFromTest(url);
```

Comment:
I would skip using creating this method and using method from props instead

Reply:
Done.